plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/service/QueryInsightsService.java

```diff
@@ -0,0 +1,300 @@
+     *
+     * @param metricType {@link MetricType}
+     * @return if the Query Insights data collection is enabled
+     */
+    public boolean isCollectionEnabled(MetricType metricType) {
+        return this.enableCollect.containsKey(metricType) && this.enableCollect.get(metricType);
+    }
+
+    /**
+     * Set the top N size for TopQueriesByLatencyService service.
+     *
+     * @param size the top N size to set
+     */
+    public void setTopNSize(int size) {
+        this.topNSize = size;
+    }
+
+    /**
+     * Validate the top N size based on the internal constrains
+     *
+     * @param size the wanted top N size
+     */
+    public void validateTopNSize(int size) {
+        if (size > QueryInsightsSettings.MAX_N_SIZE) {
+            throw new IllegalArgumentException(
+                "Top N size setting ["
+                    + QueryInsightsSettings.TOP_N_LATENCY_QUERIES_SIZE.getKey()
+                    + "]"
+                    + " should be smaller than max top N size ["
+                    + QueryInsightsSettings.MAX_N_SIZE
+                    + "was ("
+                    + size
+                    + " > "
+                    + QueryInsightsSettings.MAX_N_SIZE
+                    + ")"
+            );
+        }
+    }
+
+    /**
+     * Get the top N size set for TopQueriesByLatencyService
+     *
+     * @return the top N size
+     */
+    public int getTopNSize() {
+        return this.topNSize;
+    }
+
+    /**
+     * Set the window size for TopQueriesByLatencyService
+     *
+     * @param windowSize window size to set
+     */
+    public void setWindowSize(TimeValue windowSize) {
+        this.windowSize = windowSize;
+        for (MetricType metricType : MetricType.allMetricTypes()) {
+            topQueriesWindowStart.put(metricType, -1L);
+        }
+    }
+
+    /**
+     * Validate if the window size is valid, based on internal constrains.
+     *
+     * @param windowSize the window size to validate
+     */
+    public void validateWindowSize(TimeValue windowSize) {
+        if (windowSize.compareTo(QueryInsightsSettings.MAX_WINDOW_SIZE) > 0
+            || windowSize.compareTo(QueryInsightsSettings.MIN_WINDOW_SIZE) < 0) {
+            throw new IllegalArgumentException(
+                "Window size setting ["
+                    + QueryInsightsSettings.TOP_N_LATENCY_QUERIES_WINDOW_SIZE.getKey()
+                    + "]"
+                    + " should be between ["
+                    + QueryInsightsSettings.MAX_WINDOW_SIZE
+                    + ","
+                    + QueryInsightsSettings.MAX_WINDOW_SIZE
+                    + "]"
+                    + "was ("
+                    + windowSize
+                    + ")"
+            );
+        }
+        if (!(60 % windowSize.getMinutes() == 0 || windowSize.getMinutes() % 60 == 0)) {
+            throw new IllegalArgumentException(
+                "Window size setting ["
```

Comment:
what's the point of accepting in minutes when you're only supporting in multiple of hours?

Reply:
Actually we are supporting multiple of minutes, which can divide 60.

Reply:
Actually, let me remove this part of logic and only set constrains on min and max window sizes